dubbo-metadata-report/dubbo-metadata-report-api/src/main/java/org/apache/dubbo/metadata/integration/MetadataReportService.java

```diff
@@ -87,6 +88,9 @@ public static MetadataReportService instance(Supplier<URL> metadataReportUrl) {
     }
 
     public void publishProvider(URL providerUrl) throws RpcException {
+        if (!needStore(providerUrl)) {
+            return;
+        }
```

Comment:
If return directly, the detailed URL will also not present in metadata, that's not what we expect.